fdbserver/KeyValueStoreShardedRocksDB.actor.cpp

```diff
@@ -799,6 +908,45 @@ class ShardManager {
 		return shard;
 	}
 
+	PhysicalShard* getSpecialKeysShard() {
+		auto it = physicalShards.find(SPECIAL_KEYS_SHARD_ID);
+		ASSERT(it != physicalShards.end());
+		return it->second.get();
+	}
+
+	PhysicalShard* getMetaDataShard() {
+		auto it = physicalShards.find(METADATA_SHARD_ID);
+		ASSERT(it != physicalShards.end());
+		return it->second.get();
+	}
+
+	// Returns the physical shard that hosting all `ranges`; Returns nullptr if such a physical shard does not exists.
+	PhysicalShard* getPhysicalShardForAllRanges(std::vector<KeyRange> ranges) {
+		PhysicalShard* result = nullptr;
+		for (const auto& range : ranges) {
+			auto rangeIterator = this->dataShardMap.intersectingRanges(range);
+			for (auto it = rangeIterator.begin(); it != rangeIterator.end(); ++it) {
+				if (it.value() == nullptr || it.value()->physicalShard == nullptr) {
+					TraceEvent(SevWarn, "ShardedRocksDBRangeNotOnShard", logId).detail("Range", range);
```

Comment:
nit: suggest RangeNotInKVS or similar 
`it.value()->physicalShard == nullptr` should not happen, may add an assertion.

Reply:
Done.